src/components/clusterprovider/minikube/minikubeclusterprovider.ts

```diff
@@ -0,0 +1,224 @@
+        await this.handleRequest(getHandleCreateHtml, request, requestData, response, next);
+    }
+
+    async handleConfigure(request: restify.Request, requestData: any, response: restify.Response, next: restify.Next): Promise<void> {
+        await this.handleRequest(getHandleConfigureHtml, request, requestData, response, next);
+    }
+
+    async handleRequest(handler: HtmlRequestHandler, request: restify.Request, requestData: any, response: restify.Response, next: restify.Next) {
+        const html = await handler(request.query["step"], requestData, this.context);
+        response.contentType = 'text/html';
+        response.send("<html><body><style id='styleholder'></style>" + html + "</body></html>");
+
+        next();
+    }
+}
+
+async function getHandleCreateHtml(step: string | undefined, requestData: any, context: Context): Promise<string> {
+    if (!step) {
+        return await promptForConfiguration(requestData, context, "create", "create");
+    } else if (step === "create") {
+        return await createCluster(requestData, context);
+    } else if (step === "wait") {
+        return await waitForClusterAndReportConfigResult(context, requestData);
+    } else {
+        return renderInternalError(`MinikubeStepError (${step})`);
+    }
+}
+
+async function getHandleConfigureHtml(step: string | undefined, requestData: any): Promise<string> {
+    if (!step || step === "configure") {
+        return await configureKubernetes(requestData);
+    } else {
+        return renderInternalError(`ConfigurationStepError (${step})`);
+    }
+}
+
+async function promptForConfiguration(previousData: any, context: Context, action: clusterproviderregistry.ClusterProviderAction, nextStep: string): Promise<string> {
+    return formPage({
+        stepId: 'PromptForConfiguration',
+        title: 'Configure Minikube',
+        waitText: 'Configuring Minikube',
+        action: action,
+        nextStep: nextStep,
+        submitText: 'Create',
+        previousData: previousData,
+        formContent: `
+        <p>No configuration options (yet)</p>
+        `
+    });
+}
+
+async function configureKubernetes(previousData: any): Promise<string> {
+    const selectedCluster = previousData.cluster;
+    await refreshExplorer();
+    return renderConfigurationResult(); // TODO
+}
+
+async function runMinikubeCommand(context: Context, cmd: string): Promise<ActionResult<Diagnostic>> {
+    const sr = await context.shell.exec(cmd);
+
+    const createCluster = await fromShellExitCodeOnly(sr);
+    return {
+        actionDescription: 'creating cluster',
+        result: createCluster
+    };
+}
+
+async function createCluster(previousData: any, context: Context): Promise<string> {
+    const createResult = await runMinikubeCommand(context, 'minikube help');
+
+    context.createExec = context.shell.exec('minikube start');
+
+    const title = createResult.result.succeeded ? 'Cluster creation has started' : `Error ${createResult.actionDescription}`;
```

Comment:
This appears to be checking whether `minikube help` succeeded, not `minikube start`.  I'm not quite sure what's happening here or where `context.createExec` comes into play.

Reply:
The trouble here is that `minikube start` blocks and the current cluster provider implementation expects that cluster create is a background task.

So I need to run some command to validate that minikube is in the path, hence the check if the minikube binary is present in the user's PATH.

Reply:
Gotcha.  Let's leave this for now, then, and see if we can refine the detection later.  I'd still consider a rename of `createResult`, perhaps to `checkMinikubeExistsResult`?